### PR TITLE
Implemented new error message system

### PR DESCRIPTION
**Pull request by JackSMunn**
## Description

Implemented a new error handling system, meant to be used across the platform

## Changes

- Cleanup in all touched files
- Documentation for all touched files
- Documentation on all translations (pending proper translations from PO) both in loginForm.js and registerForm.js
- Refactored console.log in registerUser in userApi 
- Moved frontend-only validation to own file
- Added errorCodes.js, that contains all current JSON error objects
- Deleted displayPasswordGuidelines function from formTextField.js
- Deleted all traces of isFontLoaded in files
- Refactored all errors in loginForm.js to new error handling system
- Refactored registerForm to use first name and last name instead of a single name
- Added validation to name and email in registerForm.js
- Moved regex check from registerForm to validation.js
- Refactored the overall validation check (the one that checks everything) to better comply with codeScene standards for complex methods (cyclomatic complexity)
- Refactored all errors in registerForm.js to new error handling system.

## Related Issues

None documented.

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

None relevant

## Notes for Reviewers

For future reference, CERT group hopes that the rest of the teams will adopt our new error handling system.

### CODE REVIEW

File: eml/components/general/errorCodes.js
Comment (Jatewo) [inline eml/components/general/errorCodes.js:1]: We should make sure everyone uses this from the next sprint (no changes required here)

**Review by Jatewo — CHANGES_REQUESTED** [review]
### Very nice 👍 
Small changes are requested

File: eml/api/userApi.js [hunk 1]
```diff
@@ -21,23 +20,40 @@ export const client = axios.create({
 /**
  * Sends a request to the backend to register a new user.
  * @param {Object} obj Should contain the following properties:
- * - name
+ * - fistName
```
Comment (TheMOHD3886) [inline eml/api/userApi.js:23]: change to firstName

**Review by TheMOHD3886 — COMMENTED** [review]
Single typo spotted and marked with a comment. Everything else looked good.

**Pull request merged**